lib/index.js

```diff
@@ -1,12 +1,33 @@
 'use strict';
 
+var fs = require('fs');
+var path = require('path-posix');
+var assign = require('object-assign');
+var symlinkOrCopy = require('symlink-or-copy');
 var Entry = require('./entry');
 var logger = require('heimdalljs-logger')('fs-tree-diff:');
 var util = require('./util');
 var sortAndExpand = util.sortAndExpand;
 var validateSortedUnique = util.validateSortedUnique;
 
 var ARBITRARY_START_OF_TIME = 0;
+var DEFAULT_DELEGATE = {
```

Comment:
We could do that. Currently I have it throwing an error if a delegate doesn't exist, but we could probably replace that with a delegate function whose default implementation is to throw an error.